be/src/exec/workgroup/scan_executor.cpp

```diff
@@ -77,4 +78,176 @@ bool ScanExecutor::submit(ScanTask task) {
     return _task_queue->try_offer(std::move(task));
 }
 
+int ScanExecutor::submit(void* (*fn)(void*), void* args) {
+    // TODO: specify the workgroup through parameter
+    auto wg = WorkGroupManager::instance()->get_default_mv_workgroup();
+    ScanTask::WorkFunction wf = [=]() { fn(args); };
+    ScanTask task(wg.get(), std::move(wf));
+    return _task_queue->try_offer(std::move(task)) ? 0 : 1;
+}
+
+int ScanExecutor::submit(std::function<void()> fun) {
+    // TODO: specify the workgroup through parameter
+    auto wg = WorkGroupManager::instance()->get_default_mv_workgroup();
+    ScanTask task(wg.get(), std::move(fun));
+    return _task_queue->try_offer(std::move(task)) ? 0 : 1;
+}
+
+void ScanExecutor::submit_urgent(void* (*fn)(void*), void* args) {
+    _thread_pool->submit_func([=]() { fn(args); }, ThreadPool::Priority::HIGH_PRIORITY);
+}
+
+std::unique_ptr<TaskToken> ScanExecutor::new_token(const std::string& name) {
+    return std::make_unique<ExecutorToken>(name, this);
```

Comment:
Tasks in the same token is handled sequentially? if so, ExecutorToken seems to been used for memtable flushing operation in the past, memtables belongs to the same tablet flushed sequentially, different tablets' memtables are flushed parallelly.

Reply:
yes. inter-tablet parallel, inner-tablet sequential execution.

so different tablet will use different `Token`.